experimental/Schemes/SpecOpen.jl

```diff
@@ -0,0 +1,675 @@
+export SpecOpen, ambient, gens, complement, npatches, affine_patches, intersections, name, intersect, issubset, closure, find_non_zero_divisor, is_non_zero_divisor, is_dense
+
+export StructureSheafRing, scheme, domain, OO
+
+export StructureSheafElem, domain, restrictions, patches, restrict, npatches
+
+export SpecOpenMor, maps_on_patches, restriction, identity_map, preimage, generic_fractions, pullback, maximal_extension
+
+@Markdown.doc """
+    SpecOpen{BRT, BRET, RT, RET, MST} <: Scheme{BRT, BRET}
+
+Zariski open subset ``U`` of an affine scheme ``X = Spec(R)``. 
+This stores a list of generators ``f₁,…,fᵣ`` of an ideal 
+``I`` defining the complement ``Z = X ∖ U``. 
+The scheme ``X`` is referred to as the *ambient scheme* and 
+the list ``f₁,…,fᵣ`` as the *generators* for ``U``.
+"""
+@attributes mutable struct SpecOpen{BRT, BRET, RT, RET, MST} <: Scheme{BRT, BRET}
+  X::Spec{BRT, BRET, RT, RET, MST} # the ambient scheme
+  gens::Vector{RET} # a list of functions defining the complement of the open subset
```

Comment:
This is not specific to this PR, but rather to all your code on schemes so far: I have strong doubts that having all these types declared is useful. While in theory, this could allow some optimization somewhere to happen, in reality I wonder if any such places actually exist, and if they matter overall; on the downside, lots and lots of different type instances means lots and lots of functions that need to be recompiled again and again for no reason, wasting speed and memory. And then of course there is the cost by all those noisy letter salads `{BRT, BRET, RT, RET, MST}` floating around everywhere...

So, this is not for this PR, but I really think this should be reconsidered.

Reply:
See my above comment:

https://github.com/oscar-system/Oscar.jl/pull/961#discussion_r786507205